Source/Factories/StackClusterFactory.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  StackClusterFactory.swift
+//  Matrioska
+//
+//  Created by Mathias Aichinger on 11/01/2017.
+//  Copyright © 2017 runtastic. All rights reserved.
+//
+
+import Foundation
+
+extension ClusterFactory {
+    
+    /// A concrete ComponentFactory which produces a stack cluster component
+    public final class StackClusterFactory: ComponentFactory {
+        public func produce(children: [Component],
```

Comment:
Factories are meant to provide the `type` to register. Maybe we can skip that and just have JSONFactory registering the functions for a given string, but I liked the idea of representing a factory also.

In the tests I create factories for clusters and views, and I guess would also work for wrappers right?

So if I am not wrong the difference between this and the proof of concept is that here we are more explicit on the factories (which also needs more code, true) and in the proof of concept the factory only registers closures for given types.